src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -131,6 +132,17 @@ public void parseAssociatedNames(Item item, Collection<GitHubRepositoryName> res
                 for (SCM scm : triggerItem.getSCMs()) {
                     addRepositories(scm, envVars, result);
                 }
+                // additionally add the explicit GitHub project URL as defined
+                // under General job properties to the result collection
+                // as it may differ from the SCM URL due to proxy usage
+                Job job = (Job) item;
+                GithubProjectProperty ghpp = (GithubProjectProperty) job.getProperty(GithubProjectProperty.class);
+                if (ghpp != null) {
+                    GitHubRepositoryName githubrepo = GitHubRepositoryName.create(ghpp.getProjectUrl().toString());
+                    if (githubrepo != null) {
+                        result.add(githubrepo);
+                    }
+                }
```

Comment:
Should this be within the if statement for `triggerItem != null`?

Also where are the tests to test this functionality?

Reply:
This contributor is only about SCM and trigger existing in this plugin is only about SCM. 

https://github.com/KostyaSha/github-integration-plugin/ is based on GHPP and works fine without scm.